test/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/PollingNewMessagesWithWorkItemJobSpec.scala

```diff
@@ -0,0 +1,286 @@
+      await(job.executeInMutex)
+
+      verify(lockRepository).takeLock(eqTo("PollingNewMessageWithWorkItem"), any, any)
+
+      withClue("release the mongo lock") {
+        verify(lockRepository).releaseLock(eqTo("PollingNewMessageWithWorkItem"), any)
+      }
+    }
+
+    "should process workItem in ToDO status" in {
+      val workItem = createWorkItem()
+      setUpWithTwoWorkItem(workItem)
+
+      await(job.executeInMutex)
+
+      verify(workItemRepository).markAs(eqTo(workItem.id),eqTo(ProcessingStatus.ToDo), any)
+      verify(workItemRepository).complete(eqTo(workItem.id),eqTo(ProcessingStatus.Succeeded))
+    }
+
+    "catch exception Mongo throw and error" in {
+      val workItem = createWorkItem()
+      setUpWithTwoWorkItem(workItem)
+      when(workItemRepository.pullOutstanding(any, any)).thenReturn(
+        Future.failed(new RuntimeException("error"))
+      )
+
+      val result = await(job.executeInMutex)
+
+      result.message mustBe
+        """The execution of scheduled job polling-new-messages failed with error 'error'.
+          |The next execution of the job will do retry."""
+          .stripMargin
+          .replace('\n', ' ')
+    }
+
+    "retry failing EIS and mark a workItem as Failed" in {
+      val workItem = createWorkItem()
+      setUpWithTwoWorkItem(workItem)
+
+      when(newMessageService.getNewMessagesAndAcknowledge(any)(any))
+        .thenReturn(Future.failed(new RuntimeException("error")))
+
+      await(job.executeInMutex)
+
+      verify(workItemRepository, times(2)).markAs(
+        eqTo(workItem.id),
+        eqTo(ProcessingStatus.Failed),
+        any
+      )
+    }
+
+    "set the workItem state to PermanentlyFailed" in {
+      val retryAttempt = 2
+      val workItem = createWorkItem(retryAttempt)
+      setUpWithTwoWorkItem(workItem)
+      when(appConfig.retryAttempt).thenReturn(retryAttempt)
+      when(newMessageService.getNewMessagesAndAcknowledge(any)(any))
+        .thenReturn(Future.failed(new RuntimeException("error")))
+
+      await(job.executeInMutex)
+
+      verify(workItemRepository, times(2)).markAs(eqTo(workItem.id),eqTo(ProcessingStatus.PermanentlyFailed), any)
+    }
+
+    "parse the messages" in {
+      setUpWithTwoWorkItem(createWorkItem())
+
+      await(job.executeInMutex)
+
+      verify(shoeNewMessageParser).extractMessages(eqTo("any message"))
+    }
+
+    "send getNewMessage request for each pending ern" in {
+
+      setUpWithTwoWorkItem(createWorkItem())
+
+      val result = await(job.executeInMutex)
+
+      result.message mustBe "polling-new-messages Job ran successfully."
+      val captor = ArgCaptor[String]
+      verify(newMessageService, times(2)).getNewMessagesAndAcknowledge(captor.capture)(any)
+
+      captor.values mustBe Seq("123", "123")
+
+      withClue("should save message to database") {
+        verify(movementService).updateMovement(eqTo(message), eqTo("123"))
+      }
+
+    }
+
+//    "process IE801 and IE704 first " in {
```

Comment:
Is this commented out test still needed?

Reply:
removed